app/Controllers/Http/ProductsController.ts

```diff
@@ -1,38 +1,27 @@
 import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
-import products from 'App/Resourses/ProductsDummyData'
-
+import Product from 'App/Models/Product'
 
 export default class ProductsController {
-  public async index(ctx: HttpContextContract) {
-    const query = ctx.request.qs();
-    const page: number = query.page || 1;
-    const search: string = query.search || '';
-    const sortBy: string = query.sortBy || '';
-    const sortType: string = query.sortType || ''
-    let searchedProducts = products.filter((item) => {
-      return item.category.toLowerCase().includes(search.toLowerCase());
-    });
-    if (sortBy.toLowerCase() === "price") {
-      if (sortType.toLowerCase() === 'ascending') {
-        searchedProducts.sort(function (x, y) {
-          return x.price - y.price;
-        });
-      } else if (sortType.toLowerCase() === 'descending') {
-        searchedProducts.sort(function (x, y) {
-          return y.price - x.price;
-        });
-      }
-    } else if (sortBy.toLowerCase() === "title") {
-      if (sortType.toLowerCase() === 'ascending') {
-        searchedProducts.sort(function (x, y) {
-          return x.title < y.title ? -1 : 1;
-        })
-      } else if (sortType.toLowerCase() === 'descending') {
-        searchedProducts.sort(function (x, y) {
-          return x.title > y.title ? -1 : 1;
-        })
-      }
+  public async index({ request, response }: HttpContextContract) {
+    const { sortBy, sortType, page, search } = request.qs() as {
+      sortBy?: string
+      sortType?: string
+      page?: string
+      search?: string
+    }
+    if (sortType !== 'asc' && sortType !== 'desc' && sortType !== undefined) {
+      return response.badRequest('Try again')
+    }
+    let produsctsQuery = Product.query()
+    if (search) {
+      produsctsQuery = produsctsQuery.where((query) =>
+        query.where('Lower(title)', 'Like', `[%${search.toLowerCase}%]`)
```

Comment:
this is incorrect.... 
1. `toLowerCase` isn't begin called
2, The last arg is supposed to be the string... here you have given `[%<user_input>%]` this will not match with anything ever
3. first arg takes a col name .... which "Lower(title)" is not

Reply:
okk